src/XrdCl/XrdClXRootDTransport.cc

```diff
@@ -1334,53 +1376,64 @@ namespace XrdCl
     return XRootDStatus();
   }
 
-  //----------------------------------------------------------------------------
-  // Unmarshall the correction-segment of the status response for pgwrite
-  //----------------------------------------------------------------------------
-  XRootDStatus XRootDTransport::UnMarchalStatusCSE( Message &msg )
+  XRootDStatus XRootDTransport::UnMarchalStatusMore( Message &msg )
   {
     ServerResponseV2 *rsp = (ServerResponseV2*)msg.GetBuffer();
-    //--------------------------------------------------------------------------
-    // If there's no additional data there's nothing to unmarshal
-    //--------------------------------------------------------------------------
-    if( rsp->status.bdy.dlen == 0 ) return XRootDStatus();
-    //--------------------------------------------------------------------------
-    // If there's not enough data to form correction-segment report an error
-    //--------------------------------------------------------------------------
-    if( size_t( rsp->status.bdy.dlen ) < sizeof( ServerResponseBody_pgWrCSE ) )
-      return XRootDStatus( stError, errInvalidMessage, 0,
-                           "kXR_status: invalid message size." );
+    uint16_t reqType = rsp->status.bdy.requestid + kXR_1stRequest;
 
-    //--------------------------------------------------------------------------
-    // Calculate the crc32c for the additional data
-    //--------------------------------------------------------------------------
-    ServerResponseBody_pgWrCSE *cse = (ServerResponseBody_pgWrCSE*)msg.GetBuffer( sizeof( ServerResponseV2 ) );
-    cse->cseCRC = ntohl( cse->cseCRC );
-    size_t length = rsp->status.bdy.dlen - sizeof( uint32_t );
-    void*  buffer = msg.GetBuffer( sizeof( ServerResponseV2 ) + sizeof( uint32_t ) );
-    uint32_t crcval = XrdOucCRC::Calc32C( buffer, length );
-
-    //--------------------------------------------------------------------------
-    // Do the integrity checks
-    //--------------------------------------------------------------------------
-    if( crcval != cse->cseCRC )
+    switch( reqType )
```

Comment:
Why a switch with a single case inside? Do you intend this to be expanded with more cases later on?

Reply:
yes, exactly! right now `kXR_status` is used only for 'pgread'/'pgwrite' but in the future it could be used for other requests hence the switch statement.